esrally/driver/runner.py

```diff
@@ -1015,12 +1015,30 @@ async def _request_body_query(es, params):
             r = await self._raw_search(es, doc_type, index, body, request_params, headers=headers)
 
             if detailed_results:
-                props = parse(r, ["hits.total", "hits.total.value", "hits.total.relation", "timed_out", "took"])
+                props = parse(
+                    r,
+                    [
+                        "hits.total",
+                        "hits.total.value",
+                        "hits.total.relation",
+                        "timed_out",
+                        "took",
+                        "_shards.total",
+                        "_shards.successful",
+                        "_shards.skipped",
+                        "_shards.failed",
+                    ],
+                )
                 hits_total = props.get("hits.total.value", props.get("hits.total", 0))
                 hits_relation = props.get("hits.total.relation", "eq")
                 timed_out = props.get("timed_out", False)
                 took = props.get("took", 0)
 
+                shards_total = props.get("_shards.total")
```

Comment:
Should we have non-`None` defaults for these `props.get()` calls?

Reply:
Hmm, good point. I think if there's no `_shard` object in the response we should just omit it entirely, rather than populating it with a default value of `0` which could be misleading.

Reply:
Actually, I can't think of a scenario where `_shards` wouldn't be present. `0` is actually a good default.

Reply:
Addressed in https://github.com/elastic/rally/pull/1619/commits/9e9d96fffdce3af39221abc26f6d7d8c126a5769